src/main/java/com/box/sdk/BoxSearchParameters.java

```diff
@@ -0,0 +1,442 @@
+     * @return this.true;
+     */
+    public boolean clearParameters() {
+        this.query = null;
+        this.keyword = null;
+        this.fields = null;
+        this.scope = null;
+        this.fileExtensions = null;
+        this.createdAtRangeFromDate = null;
+        this.createdAtRangeToDate = null;
+        this.updatedAtRangeFromDate = null;
+        this.updatedAtRangeToDate = null;
+        this.sizeRangeLowerBoundBytes = 0;
+        this.sizeRangeUpperBoundBytes = 0;
+        this.ownerUserIds = null;
+        this.ancestorFolderIds = null;
+        this.contentTypes = null;
+        this.type = null;
+        this.trashContent = null;
+        this.mdFilters = null;
+        return true;
+    }
+    /**
+     * Get existing query String that is being used.
+     * @return this.query string.
+     */
+    public String getQuery() {
+        return this.query;
+    }
+    /**
+     * Set query string for that will be used to search.
+     * @param query is a String value.
+     */
+    public void setQuery(String query) {
+        this.query = query;
+    }
+    /**
+     * Get existing keyword String that is being used.
+     * @return this.keyword String.
+     */
+    public String getKeyword() {
+        return this.keyword;
+    }
+    /**
+     * Set keyword string for that will be used for searching.
+     * @param keyword String.
+     */
+    public void setKeyword(String keyword) {
+        this.keyword = keyword;
+    }
+    /**
+     * Get the existing fields that are used for the search criteria.
+     * @return this.List of fields.
+     */
+    public List<String> getFields() {
+        return this.fields;
+    }
+    /**
+     * Set the existing fields that are used for the search criteria.
+     * @param fields specify what fields to be returned.
+     */
+    public void setFields(List<String> fields) {
+        this.fields = fields;
+    }
+    /**
+     * Get the scope on which you want to search, ["enterprise","scope"].
+     * @return this.current scope that is set.
+     */
+    public String getScope() {
+        return this.scope;
+    }
+    /**
+     * Set the scope for how you want to search, ["enterprise","scope"].
+     * @param scope set scope you want to search.
+     */
+    public void setScope(String scope) {
+        this.scope = scope;
+    }
+    /**
+     * Get file extension filter (jpg,docx).
+     * @return this.list of extensions.
+     */
+    public List<String> getFileExtensions() {
+        return this.fileExtensions;
+    }
+    /**
+     * Set file extension by providing a list of strings [jpg,docx].
+     * @param fileExtensions applied as a filter for extensions.
+     */
+    public void setFileExtensions(List<String> fileExtensions) {
+        this.fileExtensions = fileExtensions;
+    }
+    /**
+     * Get the from date filter to specify the when a file was created.
+     * @return this.created date from.
+     */
+    public Date getCreatedAtRangeFromDate() {
```

Comment:
The names of the range methods are kind of confusing. If you create `DateRange` and `SizeRange` classes, then it'll be a bit clearer.

Usage example:

```java
DateRange range = new DateRange(from, to);
params.setCreatedAtRange(range);
```